test/javascripts/acceptance/canned-replies-test.js

```diff
@@ -1,139 +1,76 @@
-          },
-          {
-            id: "04697870e02acfef3c2130dab92fe6d8",
-            title: "Using variables",
-            excerpt:
-              "Hi %{reply_to_username,fallback:there}, regards %{my_username}.",
-            content:
-              "Hi %{reply_to_username,fallback:there}, regards %{my_username}.",
-          },
-        ],
-      });
-    });
+  needs.user({
+    can_use_canned_replies: true,
   });
+
+  needs.pretender(cannedRepliesPretender);
   needs.hooks.beforeEach(() => clearPopupMenuOptionsCallback());
 
-  test("Inserting canned replies", async (assert) => {
+  test("Filtering by tags", async (assert) => {
+    const popUpMenu = await selectKit(".toolbar-popup-menu-options");
+
     await visit("/");
 
     await click("#create-topic");
-    await fillIn(".d-editor-input", "beforeafter");
-
-    const editorInput = $(".d-editor-input")[0];
-    editorInput.selectionStart = editorInput.selectionEnd = "before".length;
-
-    const popUpMenu = await selectKit(".toolbar-popup-menu-options");
     await popUpMenu.expand();
     await popUpMenu.selectRowByValue("showCannedRepliesButton");
-    await click(".canned-reply-title");
 
-    assert.ok(
-      find(".canned-replies-content")
-        .html()
-        .indexOf("<strong>markdown</strong>") !== -1,
-      "it should display the right cooked content"
-    );
+    const tagDropdown = selectKit(".canned-replies-filter-bar .tag-drop");
+    await tagDropdown.expand();
 
-    await click(
-      "#canned-reply-ce5fc200ab90dd0d5ac597ca9bb4708b .canned-replies-apply"
+    await tagDropdown.fillInFilter(
+      "cupcake",
+      ".canned-replies-filter-bar .tag-drop input"
+    );
+    assert.deepEqual(
+      tagDropdown.displayedContent(),
+      [
+        {
+          name: "cupcakes",
+          id: "cupcakes",
+        },
+      ],
+      "it should filter tags in the dropdown"
     );
 
+    await tagDropdown.selectRowByIndex(0);
     assert.equal(
-      find(".d-editor-input").val(),
-      "before\n\n**markdown**\n\nafter",
-      "it should contain the right selected output"
+      document.querySelectorAll(".canned-replies-list .canned-reply").length,
+      1,
+      "it should filter replies by tag"
     );
-  });
 
-  test("Editing a canned reply", async (assert) => {
-    const popUpMenu = await selectKit(".toolbar-popup-menu-options");
-
-    await visit("/");
-
-    await click("#create-topic");
-    await popUpMenu.expand();
-    await popUpMenu.selectRowByValue("showCannedRepliesButton");
-
-    await click(".canned-replies-edit");
-
-    await fillIn(".canned-replies-form-title-input", "Some title");
-    await fillIn(".canned-replies-form-content-input textarea", "Some content");
-
-    await click(".edit-reply-save-btn");
+    await click("#canned-reply-1 .canned-replies-apply");
 
     assert.equal(
-      find(".canned-replies-footer .edit-reply-save-btn").text().trim(),
-      I18n.t("saved")
+      find(".d-editor-input").val().trim(),
```

Comment:
`find` is about to be deprecated. `query()` helper from `discourse/tests/helpers/qunit-helpers` is preferred (it doesn't use jQuery):

```suggestion
      query(".d-editor-input").value.trim(),
```

(the same applies to the test below)

Reply:
Fixed in a2627cd602f399296ff639ddecc17f17d94e01c3